src/Capture/CaptureMode.php

```diff
@@ -0,0 +1,58 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Academic Free License (AFL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/AFL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/AFL-3.0  Academic Free License (AFL 3.0)
+ */
+
+namespace PrestaShop\Module\PrestashopCheckout\Capture;
+
+/**
+ * Class CaptureMode used to get / update the capture mode from configuration
+ */
+class CaptureMode
```

Comment:
This is related to PayPal Intent and CAPTURE is only one possible value.
It can be AUTHORIZE too
So I thinks it must be renamed to `Intent` and placed in namespace `PrestaShop\Module\PrestashopCheckout\PayPal`

Moreover, if we want to stay consistent with [LoggerConfiguration](https://github.com/Matt75/ps_checkout/blob/6976f8b4b58e02243bd67a1575f096b76f45e4e3/src/Logger/LoggerConfiguration.php) service
I thinks we can create a `PayPalConfiguration` service can be responsible to interact with `PrestaShopConfiguration` service and can have `getIntent()` and `setIntent()` in order to retrieve more trusted values (right type etc...) than retrieve only from PrestaShopConfiguration directly with mixed return type.

Reply:
Agree with this, i will do that so ;)

Reply:
done in next commit